src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -0,0 +1,241 @@
+import com.faforever.client.map.generator.MapGeneratorService;
+import com.faforever.client.mod.FeaturedMod;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.preferences.GeneratorPrefs;
+import com.faforever.client.preferences.PreferenceUpdateListener;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.annotations.VisibleForTesting;
+import com.jfoenix.controls.JFXButton;
+import com.jfoenix.controls.JFXTextField;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.transformation.FilteredList;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.MultipleSelectionModel;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.Slider;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.SpinnerValueFactory.IntegerSpinnerValueFactory;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.Pane;
+import javafx.util.StringConverter;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.ref.WeakReference;
+import java.util.Objects;
+import java.util.Random;
+import java.util.function.Function;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+public class GenerateMapController implements Controller<Pane> {
+
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final PreferencesService preferencesService;
+  private final NotificationService notificationService;
+  private final MapGeneratorService mapGeneratorService;
+  public CreateGameController createGameController;
+  public Pane generateMapRoot;
+  public JFXButton generateMapButton;
+  public JFXTextField previousMapName;
+  public Spinner<Integer> spawnCountSpinner;
+  public Slider waterSlider;
+  public CheckBox waterRandom;
+  public Slider mountainSlider;
+  public CheckBox mountainRandom;
+  public Slider plateauSlider;
+  public CheckBox plateauRandom;
+  public Slider rampSlider;
+  public CheckBox rampRandom;
+  @VisibleForTesting
+  FilteredList<MapBean> filteredMapBeans;
+  private Runnable onCloseButtonClickedListener;
+  private PreferenceUpdateListener preferenceUpdateListener;
+
+  /**
+   * Remembers if the controller's init method was called, to avoid memory leaks by adding several listeners
+   */
+  private boolean initialized;
+
+  public void initialize() {
+    if (preferencesService.getPreferences().getForgedAlliance().getInstallationPath() == null) {
+      preferenceUpdateListener = preferences -> {
+        if (!initialized && preferencesService.getPreferences().getForgedAlliance().getInstallationPath() != null) {
```

Comment:
Are you sure this check is not already made by the game and replay services? It should be imo. Have you tried removing this and removing the installation setting and the trying to play?

Reply:
Removed